### PR TITLE
[Stableswap Simulator] add generic Plot component with default layout and config

### PR DESCRIPTION
**Pull request by yvesfracari**
Use example:

image

### CODE REVIEW

File: packages/next-app/src/components/Plot.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  useResizeHandler: true,
+  layout: {
+    plot_bgcolor: blueDarkA.blueA1,
+    paper_bgcolor: blueDarkA.blueA1,
+    font: {
+      color: grayDarkA.grayA12,
+      family: "Inter",
+    },
+    xaxis: JSON.parse(JSON.stringify(defaultAxisLayout)),
+    yaxis: JSON.parse(JSON.stringify(defaultAxisLayout)),
```
Comment (ribeirojose) [inline packages/next-app/src/components/Plot.tsx:24]: 👀 care to enlighten me in why would we need this here?

File: packages/next-app/src/components/Plot.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    showEditInChartStudio: false,
+    showLink: false,
+    watermark: false,
+  },
+  revision: 0,
+};
+
+export default function Plot(props: PlotParams) {
+  const plotProps = merge(JSON.parse(JSON.stringify(defaultPlotProps)), props);
+  return <PlotRoot {...plotProps} key={JSON.stringify(plotProps)} />;
```
Comment (ribeirojose) [inline packages/next-app/src/components/Plot.tsx:46]: I'd suggest letting the caller define a key here -- probably not an issue now but plots with the same props would probably lead to a bug in how plotly is handling these plots

File: packages/next-app/src/components/Plot.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    showSendToCloud: false,
+    showEditInChartStudio: false,
+    showLink: false,
+    watermark: false,
+  },
+  revision: 0,
+};
+
+export default function Plot(props: PlotParams) {
+  const plotProps = merge(JSON.parse(JSON.stringify(defaultPlotProps)), props);
```
Comment (ribeirojose) [inline packages/next-app/src/components/Plot.tsx:45]: this is very odd and prone to WTFs by newcomers -- if we must do it please leave a comment here so whomever understands why this is needed

**Review by ribeirojose — CHANGES_REQUESTED** [review]
only `#must` is address the subsequent json.parse/json.stringify calls. That smells like it's covering into another issue and can very quickly lead to spaghetti code. We must add a comment explaining it or fix whatever the underlying issue is.

File: packages/next-app/src/components/Plot.tsx (hunk 3 above)
Comment (yvesfracari) [reply packages/next-app/src/components/Plot.tsx:45]: This is needed bc the merge changes the first object (defaultPlotProps) with the function result. So, I am using the json.parse/json.stringify to deep copy the defaultPlotProps

**Review by ribeirojose — DISMISSED** [review]

**Pull request merged**